### ISSUE TITLE
Canonical tags on listings

### ISSUE THREAD

**Issue by figart** [digett-trs/trs#36]
When a user clicks on a listing card, they should see not only the listing, but also the URL in the browser bar should change (according to the slug). There are two versions (views) of the listing page: _modal_ and _non-modal_.

The _modal_ view is what a user sees when clicking on the listing card from the home page. It preserves the sorting/filtering results on the home page while the user inspects the details of a listing. It's URL is distinguished from the non-modal view by a URL parameter (e.g., https://www.texasranchesforsale.com/hidden-valley-ranch-medina-county/?isModal=true)

The _non-modal_ view serves as the canonical URL for the listing, improving search indexing by eliminating the home page content from a search engine's crawl. This view is presented like a normal page on the website, using the site's header and footer section.

As implied in the description of the modal view, the page's canonical reference should be set with a rel="canonical" tag. This tag must appear with the modal view, and ideally should also appear in the non-modal view. The canonical URL will always be the URL of the listing _without_ the URL parameter.

**Comment by emontes** [comment]
What wil be the canonical url?  I mean wil be https://www.texasranchesforsale.com/ ?

**Comment by emontes** [comment]
Solved, now it's showing the canonical href

![image](https://user-images.githubusercontent.com/1318017/195486148-15cab997-1ec1-475a-95a9-1bec7d497d30.png)

**Comment by emontes** [comment]
Yes there is a way,  and it's done :)

**Comment by emontes** [comment]
> 

Well after all we can't
``` 
//const host = window.location.protocol + "//" + window.location.host
  const host = "https://trs-development.gtsb.io"
```

If I use te variable host as window.location....etc it causes error in build time with ssr (Server Side Rendering). So unfortunately we do have to leave it as a fixed value.

**Comment by figart** [comment]
Okay, what if we do this, then... can we set this value in Sanity? We could add the value to this doctype and document: https://trs.sanity.studio/production/desk/siteSettings;settings

**Comment by emontes** [comment]
- can we set this value in Sanity?

I don't think, so, it's better to handle it in the gatsby-config file, however if you do a domain change or something like that, it'll have to be done by a developer.

![image](https://user-images.githubusercontent.com/1318017/195893823-9a4f6766-0b0d-4010-93b1-4fb5f0ede6fe.png)

Here is the section in the gatsby-config, I will add a variable called url (or something like that)

**Comment by digget** [comment]
@emontes i think you mentioned the wrong Person 

Greetings

**Comment by figart** [comment]
> better to handle it in the gatsby-config file

Ah, yes, I'm totally good with that. 

> missing a value for "Description"

I just grabbed this from the legacy site: Texas Ranches for Sale is a ranch brokerage firm that works with both buyers and sellers using cutting edge technology for marketing and locating South Texas Ranches, Hill Country Ranches, West Texas Ranches and Beyond.

> the variable 'author' basically is used for the twitter card

Can we just eliminate that variable? I'm not sure how useful it is.

My apologies... I'd typed this out a couple days ago and only now am finding that I failed to submit.

**Comment by emontes** [comment]
- Adjusted use of url insed siteMetadata in gatsby-config.js
- Added value to default description
- Removed value for Author in gatsby-config.js

**Comment by figart** [comment]
@emontes this looks good in modal property view, but I think the non-modal view is _not_ using your siteMetadata in the gatsby config, as it uses a different hostname. Here are the canonical values I found for the same property in modal vs. non-modal view:

- https://trs.gatsbyjs.io/property/sutton-county-l5-ranch-w-of-junction-ranch-for-sale
- https://trs-development.gtsb.io/property/sutton-county-l5-ranch-w-of-junction-ranch-for-sale

**Comment by emontes** [comment]
I think https://trs-development.gtsb.io/property/sutton-county-l5-ranch-w-of-junction-ranch-for-sale is comming from the old version.  Now we are deploying to  trs.gatsbyjs.io (not to trs-development.gtsb.io)

**Comment by digett-trs** [comment]
Okay, maybe I confused myself before... I'm going to leave this one open but reassign to myself so that i can check it again later.

**Comment by figart** [comment]
@emontes, am just asking the question here... how much trouble would it be to move this value out of gatsby-config.js and replace with an environment variable? I just don't love the idea of having this buried somewhere in code.

**Comment by emontes** [comment]
Yes It's possible, I have already do something similar for the globalSharing image. 
```
Note:  For sending the changes to Sanity server I have to use:
sanity deploy
An for GraphQl notice the change I have to use:
sanity graphql deploy
```
Note: Code taken from 
https://github.com/digett-trs/trs/issues/51#issuecomment-1292621600

Solved.
Finally all the Seo data is taken from Sanity, also the title and description.


Aditional question: figart and digett-trs are the same person (Mark)?  No big deal, just curiosity :)

**Comment by figart** [comment]
I'm confused on this one, @emontes. I'm not sure what you've changed. Let's hop on Discord tomorrow and discuss. I'll leave this assigned to me for now.
